proposed/clock-meta.md

```diff
@@ -0,0 +1,146 @@
+# Clock Meta Document
+
+## 1. Summary
+
+The purpose of using the `ClockInterface` is to provide a standard way to access the system 
+time, that would allow interopability when testing code that relies on the current time 
+rather than relying on installing PHP extensions or use hacks like re-declaring the `time()`
+function in other namespaces. 
+
+## 2. Why Bother?
+
+There are currently a few libraries that do provide this functionality on packagist, however 
+there is no interopability between these different libraries, as they ship with their own 
+clock interfaces. Symfony provides a package callsed `symfony/phpunit-bridge` that has a
+`Symfony\Bridge\PhpUnit\ClockMock` class, which allows mocking PHP's built-in time & date 
+functions, however this does not solve mocking calls to `new \DateTimeImmutable()`. It does
+not fully mock time when called from other libraries that rely on the system time. 
+`Cake\Chronos\Chronos` does provide mocking, however it is set via a global (static class 
+property), and this has its own pitfalls as it provides no isolation.
+
+Pros:
+
+* Consistent interface to get the current time;
+* Easy to mock the wall clock time for repeatablility.
+
+Cons:
+
+* Extra overhead and developer effort to get the current time, not as simple as
+calling `time()` or `date()`.
+
+## 3. Scope
+
+### 3.1 Goals
+
+* Provide a simple and mockable way to read the current time;
+* Allow interoperability between libraries when reading the clock.
```

Comment:
Not sure, do we need a trailing `;` or `.` here? If so, perhaps consistently use one?

Reply:
Not sure, perhaps @ashnazg can weight in on this?